mlflow/server/js/src/experiment-tracking/components/ExperimentPage.js

```diff
@@ -135,12 +163,27 @@ export class ExperimentPage extends Component {
     this.handleGettingRuns(this.props.searchRunsApi, this.state.searchRunsRequestId);
   }
 
-  maybeReloadData(prevProps) {
+  maybeReloadData(prevProps, prevState) {
     if (this.props.experimentId !== prevProps.experimentId) {
       this.loadData();
+    } else if (this.filtersDidUpdate(prevState)) {
```

Comment:
Do we need to check for all filters or can we just check if the URL state changed?

Reply:
Some url state changes do not require a trip to the database (such as a column being unchecked, or the toggle between multicolumn view and regular view). Therefore, we check if any filters have changed that do require a trip to the database before calling `handleGettingRuns`